src/common/HeartbeatMap.cc

```diff
@@ -120,9 +120,11 @@ bool HeartbeatMap::is_healthy()
   int total = 0;
   m_rwlock.get_read();
   auto now = ceph::coarse_mono_clock::now();
-  if (m_cct->_conf->heartbeat_inject_failure) {
-    ldout(m_cct, 0) << "is_healthy injecting failure for next " << m_cct->_conf->heartbeat_inject_failure << " seconds" << dendl;
-    m_inject_unhealthy_until = now + std::chrono::seconds(m_cct->_conf->heartbeat_inject_failure);
+  auto hb_inject_failure =
+    m_cct->_conf->get_val<int64_t>("heartbeat_inject_failure");
```

Comment:
could you take this chance to remove the entry from `legacy_config_opts.h` and update `options.cc`, with its description for instance.